src/app/(proper_react)/(redesign)/(public)/layout.tsx

```diff
@@ -7,29 +7,50 @@ import { PublicShell } from "./PublicShell";
 import {
   getAcceptLangHeaderInServerComponents,
   getL10n,
+  getL10nBundles,
 } from "../../../functions/l10n/serverComponents";
 import { headers } from "next/headers";
 import { getCountryCode } from "../../../functions/server/getCountryCode";
 import { getLocale } from "../../../functions/universal/getLocale";
 import { getExperiments } from "../../../functions/server/getExperiments";
 import { getExperimentationId } from "../../../functions/server/getExperimentationId";
 import { getEnabledFeatureFlags } from "../../../../db/tables/featureFlags";
+import { getServerSession } from "../../../functions/server/getServerSession";
+import * as Sentry from "@sentry/nextjs";
 
 export default async function Layout(props: { children: ReactNode }) {
-  const headersList = await headers();
-  const countryCode = getCountryCode(headersList);
   const enabledFeatureFlags = await getEnabledFeatureFlags({
     isSignedOut: true,
   });
   const l10n = getL10n(await getAcceptLangHeaderInServerComponents());
-  const experimentationId = await getExperimentationId(null);
+  const currentLocale = getLocale(
+    getL10nBundles(await getAcceptLangHeaderInServerComponents()),
+  );
+  const session = await getServerSession();
+  const headersList = await headers();
+  const countryCode = getCountryCode(headersList);
+
+  // Check for Nimbus preview mode. Note that this requires a full page reload
+  // to activate: https://nextjs.org/docs/app/api-reference/file-conventions/layout#caveats
+  const nimbusPreviewMode = (await headers()).get("x-nimbus-preview-mode");
+  const experimentationId = await getExperimentationId(session?.user ?? null);
   const experimentData = await getExperiments({
-    experimentationId,
-    countryCode,
-    locale: getLocale(l10n),
-    previewMode: false,
+    experimentationId: experimentationId,
+    countryCode: countryCode,
+    locale: currentLocale,
+    previewMode: nimbusPreviewMode === "true",
   });
 
+  const nimbus_user_id = experimentData["Enrollments"]?.nimbus_user_id;
```

Comment:
The code you copied here unfortunately was incorrect; `.Enrollments` is an array, not an object. I have a fix here: https://github.com/mozilla/blurts-server/pull/5509/commits/7d5da75ef7cdfb3b4f6167403393abfba736ba4c